File: server-scala-play-server/app/api/BillableEntityApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.BillableEntityResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class BillableEntityApiController @Inject()(cc: ControllerComponents, api: BillableEntityApi) extends AbstractController(cc) {
  /**
    * POST /api/3.18/billable/create?deviceId=[value]&accountId=[value]&name=[value]&streetAddress=[value]&streetAddress2=[value]&city=[value]&state=[value]&postalCode=[value]&businessPhone=[value]&businessPhoneExt=[value]&authorizeNetApiKey=[value]&authorizeNetTransactionKey=[value]
    */
  def createBillableEntity(): Action[AnyContent] = Action { request =>
    def executeApi(): BillableEntityResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val name = request.getQueryString("name")
        
      val streetAddress = request.getQueryString("streetAddress")
        
      val streetAddress2 = request.getQueryString("streetAddress2")
        
      val city = request.getQueryString("city")
        
      val state = request.getQueryString("state")
        
      val postalCode = request.getQueryString("postalCode")
        
      val businessPhone = request.getQueryString("businessPhone")
        
      val businessPhoneExt = request.getQueryString("businessPhoneExt")
        
      val authorizeNetApiKey = request.getQueryString("authorizeNetApiKey")
        
      val authorizeNetTransactionKey = request.getQueryString("authorizeNetTransactionKey")
        
      api.createBillableEntity(deviceId, accountId, name, streetAddress, streetAddress2, city, state, postalCode, businessPhone, businessPhoneExt, authorizeNetApiKey, authorizeNetTransactionKey)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/billable/delete?deviceId=[value]&accountId=[value]
    */
  def deleteBillableEntity(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      api.deleteBillableEntity(deviceId, accountId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/billable/get?deviceId=[value]&accountId=[value]&includeCounts=[value]&includePayments=[value]
    */
  def getBillableEntity(): Action[AnyContent] = Action { request =>
    def executeApi(): BillableEntityResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val includeCounts = request.getQueryString("includeCounts")
        .map(value => value.toBoolean)
        
      val includePayments = request.getQueryString("includePayments")
        .map(value => value.toBoolean)
        
      api.getBillableEntity(deviceId, accountId, includeCounts, includePayments)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/billable/update?deviceId=[value]&accountId=[value]&name=[value]&streetAddress=[value]&streetAddress2=[value]&city=[value]&state=[value]&postalCode=[value]&businessPhone=[value]&businessPhoneExt=[value]&authorizeNetApiKey=[value]&authorizeNetTransactionKey=[value]
    */
  def updateBillableEntity(): Action[AnyContent] = Action { request =>
    def executeApi(): BillableEntityResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val name = request.getQueryString("name")
        
      val streetAddress = request.getQueryString("streetAddress")
        
      val streetAddress2 = request.getQueryString("streetAddress2")
        
      val city = request.getQueryString("city")
        
      val state = request.getQueryString("state")
        
      val postalCode = request.getQueryString("postalCode")
        
      val businessPhone = request.getQueryString("businessPhone")
        
      val businessPhoneExt = request.getQueryString("businessPhoneExt")
        
      val authorizeNetApiKey = request.getQueryString("authorizeNetApiKey")
        
      val authorizeNetTransactionKey = request.getQueryString("authorizeNetTransactionKey")
        
      api.updateBillableEntity(deviceId, accountId, name, streetAddress, streetAddress2, city, state, postalCode, businessPhone, businessPhoneExt, authorizeNetApiKey, authorizeNetTransactionKey)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/BillableEntityApiController.java
package controllers;

import java.math.BigDecimal;
import apimodels.BillableEntityResponse;
import apimodels.SirqulResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class BillableEntityApiController extends Controller {
    private final BillableEntityApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private BillableEntityApiController(Config configuration, BillableEntityApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result createBillableEntity(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valuename = request.getQueryString("name");
        String name;
        if (valuename != null) {
            name = valuename;
        } else {
            name = null;
        }
        String valuestreetAddress = request.getQueryString("streetAddress");
        String streetAddress;
        if (valuestreetAddress != null) {
            streetAddress = valuestreetAddress;
        } else {
            streetAddress = null;
        }
        String valuestreetAddress2 = request.getQueryString("streetAddress2");
        String streetAddress2;
        if (valuestreetAddress2 != null) {
            streetAddress2 = valuestreetAddress2;
        } else {
            streetAddress2 = null;
        }
        String valuecity = request.getQueryString("city");
        String city;
        if (valuecity != null) {
            city = valuecity;
        } else {
            city = null;
        }
        String valuestate = request.getQueryString("state");
        String state;
        if (valuestate != null) {
            state = valuestate;
        } else {
            state = null;
        }
        String valuepostalCode = request.getQueryString("postalCode");
        String postalCode;
        if (valuepostalCode != null) {
            postalCode = valuepostalCode;
        } else {
            postalCode = null;
        }
        String valuebusinessPhone = request.getQueryString("businessPhone");
        String businessPhone;
        if (valuebusinessPhone != null) {
            businessPhone = valuebusinessPhone;
        } else {
            businessPhone = null;
        }
        String valuebusinessPhoneExt = request.getQueryString("businessPhoneExt");
        String businessPhoneExt;
        if (valuebusinessPhoneExt != null) {
            businessPhoneExt = valuebusinessPhoneExt;
        } else {
            businessPhoneExt = null;
        }
        String valueauthorizeNetApiKey = request.getQueryString("authorizeNetApiKey");
        String authorizeNetApiKey;
        if (valueauthorizeNetApiKey != null) {
            authorizeNetApiKey = valueauthorizeNetApiKey;
        } else {
            authorizeNetApiKey = null;
        }
        String valueauthorizeNetTransactionKey = request.getQueryString("authorizeNetTransactionKey");
        String authorizeNetTransactionKey;
        if (valueauthorizeNetTransactionKey != null) {
            authorizeNetTransactionKey = valueauthorizeNetTransactionKey;
        } else {
            authorizeNetTransactionKey = null;
        }
        return imp.createBillableEntityHttp(request, version, deviceId, accountId, name, streetAddress, streetAddress2, city, state, postalCode, businessPhone, businessPhoneExt, authorizeNetApiKey, authorizeNetTransactionKey);
    }

    @ApiAction
    public Result deleteBillableEntity(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        return imp.deleteBillableEntityHttp(request, version, deviceId, accountId);
    }

    @ApiAction
    public Result getBillableEntity(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueincludeCounts = request.getQueryString("includeCounts");
        Boolean includeCounts;
        if (valueincludeCounts != null) {
            includeCounts = Boolean.valueOf(valueincludeCounts);
        } else {
            includeCounts = false;
        }
        String valueincludePayments = request.getQueryString("includePayments");
        Boolean includePayments;
        if (valueincludePayments != null) {
            includePayments = Boolean.valueOf(valueincludePayments);
        } else {
            includePayments = true;
        }
        return imp.getBillableEntityHttp(request, version, deviceId, accountId, includeCounts, includePayments);
    }

    @ApiAction
    public Result updateBillableEntity(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valuename = request.getQueryString("name");
        String name;
        if (valuename != null) {
            name = valuename;
        } else {
            name = null;
        }
        String valuestreetAddress = request.getQueryString("streetAddress");
        String streetAddress;
        if (valuestreetAddress != null) {
            streetAddress = valuestreetAddress;
        } else {
            streetAddress = null;
        }
        String valuestreetAddress2 = request.getQueryString("streetAddress2");
        String streetAddress2;
        if (valuestreetAddress2 != null) {
            streetAddress2 = valuestreetAddress2;
        } else {
            streetAddress2 = null;
        }
        String valuecity = request.getQueryString("city");
        String city;
        if (valuecity != null) {
            city = valuecity;
        } else {
            city = null;
        }
        String valuestate = request.getQueryString("state");
        String state;
        if (valuestate != null) {
            state = valuestate;
        } else {
            state = null;
        }
        String valuepostalCode = request.getQueryString("postalCode");
        String postalCode;
        if (valuepostalCode != null) {
            postalCode = valuepostalCode;
        } else {
            postalCode = null;
        }
        String valuebusinessPhone = request.getQueryString("businessPhone");
        String businessPhone;
        if (valuebusinessPhone != null) {
            businessPhone = valuebusinessPhone;
        } else {
            businessPhone = null;
        }
        String valuebusinessPhoneExt = request.getQueryString("businessPhoneExt");
        String businessPhoneExt;
        if (valuebusinessPhoneExt != null) {
            businessPhoneExt = valuebusinessPhoneExt;
        } else {
            businessPhoneExt = null;
        }
        String valueauthorizeNetApiKey = request.getQueryString("authorizeNetApiKey");
        String authorizeNetApiKey;
        if (valueauthorizeNetApiKey != null) {
            authorizeNetApiKey = valueauthorizeNetApiKey;
        } else {
            authorizeNetApiKey = null;
        }
        String valueauthorizeNetTransactionKey = request.getQueryString("authorizeNetTransactionKey");
        String authorizeNetTransactionKey;
        if (valueauthorizeNetTransactionKey != null) {
            authorizeNetTransactionKey = valueauthorizeNetTransactionKey;
        } else {
            authorizeNetTransactionKey = null;
        }
        return imp.updateBillableEntityHttp(request, version, deviceId, accountId, name, streetAddress, streetAddress2, city, state, postalCode, businessPhone, businessPhoneExt, authorizeNetApiKey, authorizeNetTransactionKey);
    }

}
